### ISSUE TITLE
show chassis alarm : issue

### ISSUE THREAD

**Issue by door7302** [Juniper/jsnapy#324]
### Description of Issue/Question

I experienced an issue to track changes (new alarm, alarm was cleared...) of show chassis alarms between pre and post snapshot. 
My jsnapy test template below works as expected when in I still have at least one alarm in pre and post snap. If in the pre snap or post snap there is no more alarm then there is an error : due to missing XPATH. 

How to accomplish a simple check of chassis alarms ? 

### Setup
```yaml
tests_include:
  - chassis_alarms

chassis_alarms:
  - command: show chassis alarms
  - iterate:
      xpath: alarm-detail
      id: alarm-description
      tests:
        - list-not-less: alarm-short-description
          ignore-null: True
          err: "This alarm has disappeared {{pre['alarm-description']}}"
        - list-not-more: alarm-short-description
          ignore-null: True
          err: "A new alarm is present {{post['alarm-description']}}"
```
### Steps to Reproduce Issue
jsnapy --snap pre -f config.yml -v

cat /root/jsnapy/snapshots/myrouter_pre_show_chassis_alarms.xml
```xml
<alarm-information>
<alarm-summary>
<no-active-alarms/>
</alarm-summary>
</alarm-information>
```
jsnapy --snap post -f config.yml -v
cat /root/jsnapy/snapshots/myrouter_post_show_chassis_alarms.xml
```xml
<alarm-information>
<alarm-summary>
<active-alarm-count>1</active-alarm-count>
</alarm-summary>
<alarm-detail>
<alarm-time seconds="1532001817">
2018-07-19 14:03:37 CEST
</alarm-time>
<alarm-class>Minor</alarm-class>
<alarm-description>Plane 0 not online</alarm-description>
<alarm-short-description>Plane 0 not online</alarm-short-description>
<alarm-type>Chassis</alarm-type>
</alarm-detail>
</alarm-information>
```

jsnapy --check pre post -f config.yml -v
```
*************************** Device: 81.253.192.16 ***************************
Tests Included: chassis_alarms
************************ Command: show chassis alarms ************************
-------------------Performing list-not-less Test Operation-------------------
SKIPPING!! Nodes are not present in given Xpath: <alarm-detail>
-------------------Performing list-not-more Test Operation-------------------
SKIPPING!! Nodes are not present in given Xpath: <alarm-detail>
------------------------------- Final Result!! -------------------------------
chassis_alarms : Skipped
Total No of tests passed: 0
Total No of tests failed: 0
None of the test cases executed !!!
```
### Versions Report
JSNAPy version: 1.3.1

**Comment by rahkumar651991** [comment]
Fixed along with 9ad8375f04bf9eb2bd2283fac2260753e6f51949.

Testing done for it is as mentioned below.

chasis_alarm.yml

chassis_alarms:
  - command: show chassis alarms
  - iterate:
      xpath: alarm-detail
      id: alarm-description
      tests:
        - list-not-less: alarm-short-description
          ignore-null: True
          err: "This alarm has disappeared {{pre['alarm-description']}}"
        - list-not-more: alarm-short-description
          ignore-null: True
          err: "A new alarm is present {{post['alarm-description']}}"

Testcase 1

cat ~/jsnapy/snapshots/router_pre_show_chassis_alarms.xml

< alarm-information>
< alarm-summary>
< active-alarm-count>1</active-alarm-count>
< /alarm-summary>
< alarm-detail>
< alarm-time seconds="1532001817">
2018-07-19 14:03:37 CEST
< /alarm-time>
< alarm-class>Minor</alarm-class>
< alarm-description>Plane 0 not online</alarm-description>
< alarm-short-description>Plane 0 not online</alarm-short-description>
< alarm-type>Chassis</alarm-type>
< /alarm-detail>
< /alarm-information>

cat ~/jsnapy/snapshots/router_post_show_chassis_alarms.xml

< alarm-information>
< alarm-summary>
< active-alarm-count>2</active-alarm-count>
< /alarm-summary>
< alarm-detail>
< alarm-time seconds="1532001817">
2018-07-19 14:03:37 CEST
< /alarm-time>
< alarm-class>Minor</alarm-class>
< alarm-description>Plane 0 not online</alarm-description>
< alarm-short-description>Plane 0 not online</alarm-short-description>
< alarm-type>Chassis</alarm-type>
< /alarm-detail>
< alarm-detail>
< alarm-time seconds="1532001820">
2018-07-19 14:03:39 CEST
< /alarm-time>
< alarm-class>Minor</alarm-class>
< alarm-description>Plane 1 not online</alarm-description>
< alarm-short-description>Plane 1 not online</alarm-short-description>
< alarm-type>Chassis</alarm-type>
< /alarm-detail>
< /alarm-information>


jsnapy --check pre post -f test.yml -v

jsnapy.cfg file location used : /etc/jsnapy
Configuration file location used : /etc/jsnapy
*************************** Device: router ***************************
Tests Included: chassis_alarms
************************ Command: show chassis alarms ************************
-------------------Performing list-not-less Test Operation-------------------
Test PASSED: alarm-short-description before was < Plane 0 not online > now it is < Plane 0 not online >
PASS | All "alarm-short-description" in pre snapshot is present in post snapshot [ 1 matched ]
-------------------Performing list-not-more Test Operation-------------------
Test PASSED: alarm-short-description before was < Plane 0 not online > now it is < Plane 0 not online >
PASS | All "alarm-short-description" in pre snapshot is present in post snapshot [ 1 matched ]
ID gone missing!!

ID list ' {'id_0': ['Plane 1 not online']} ' is not present in pre snapshots
A new alarm is present Plane 1 not online
FAIL | All "alarm-short-description" in post snapshot is not present in pre snapshot [ 1 matched / 1 failed ]
------------------------------- Final Result!! -------------------------------
chassis_alarms : Failed
Total No of tests passed: 1
Total No of tests failed: 1
Overall Tests failed!!!

Testcase 2

cat ~/jsnapy/snapshots/router_pre_show_chassis_alarms.xml

< alarm-information>
< alarm-summary>
< active-alarm-count>1</active-alarm-count>
< /alarm-summary>
< alarm-detail>
< alarm-time seconds="1532001817">
2018-07-19 14:03:37 CEST
< /alarm-time>
< alarm-class>Minor</alarm-class>
< alarm-description>Plane 0 not online</alarm-description>
< alarm-short-description>Plane 0 not online</alarm-short-description>
<alarm-type>Chassis</alarm-type>
</alarm-detail>
</alarm-information>

cat ~/jsnapy/snapshots/router_post_show_chassis_alarms.xml

< alarm-information>
< alarm-summary>
< active-alarm-count>2</active-alarm-count>
< /alarm-summary>
< alarm-detail>
< alarm-time seconds="1532001817">
2018-07-19 14:03:37 CEST
< /alarm-time>
< alarm-class>Minor</alarm-class>
< alarm-description>Plane 0 not online</alarm-description>
< alarm-short-description>Plane 0 not online</alarm-short-description>
< alarm-type>Chassis</alarm-type>
< /alarm-detail>
< /alarm-information>


jsnapy --check pre post -f test.yml -v

jsnapy.cfg file location used : /etc/jsnapy
Configuration file location used : /etc/jsnapy
*************************** Device: router ***************************
Tests Included: chassis_alarms
************************ Command: show chassis alarms ************************
-------------------Performing list-not-less Test Operation-------------------
Test PASSED: alarm-short-description before was < Plane 0 not online > now it is < Plane 0 not online >
PASS | All "alarm-short-description" in pre snapshot is present in post snapshot [ 1 matched ]
-------------------Performing list-not-more Test Operation-------------------
Test PASSED: alarm-short-description before was < Plane 0 not online > now it is < Plane 0 not online >
PASS | All "alarm-short-description" in post snapshot is present in pre snapshot [ 1 matched ]
------------------------------- Final Result!! -------------------------------
chassis_alarms : Passed
Total No of tests passed: 2
Total No of tests failed: 0
Overall Tests passed!!!




Testcase 3

cat ~/jsnapy/snapshots/router_pre_show_chassis_alarms.xml

< alarm-information>
< alarm-summary>
< no-active-alarms/>
< /alarm-summary>
< /alarm-information>

cat ~/jsnapy/snapshots/router_post_show_chassis_alarms.xml

< alarm-information>
< alarm-summary>
< active-alarm-count>2</active-alarm-count>
< /alarm-summary>
< alarm-detail>
< alarm-time seconds="1532001817">
2018-07-19 14:03:37 CEST
< /alarm-time>
< alarm-class>Minor</alarm-class>
< alarm-description>Plane 0 not online</alarm-description>
< alarm-short-description>Plane 0 not online</alarm-short-description>
< alarm-type>Chassis</alarm-type>
< /alarm-detail>
< /alarm-information>

jsnapy --check pre post -f test.yml -v

jsnapy.cfg file location used : /etc/jsnapy
Configuration file location used : /etc/jsnapy
*************************** Device: router ***************************
Tests Included: chassis_alarms
************************ Command: show chassis alarms ************************
-------------------Performing list-not-less Test Operation-------------------
SKIPPING!! Nodes are not present in given Xpath: <alarm-detail>
-------------------Performing list-not-more Test Operation-------------------
ID gone missing!!

ID list ' {'id_0': ['Plane 0 not online']} ' is not present in pre snapshots
A new alarm is present Plane 0 not online
FAIL | All "alarm-short-description" in post snapshot is not present in pre snapshot [ 0 matched / 1 failed ]
------------------------------- Final Result!! -------------------------------
chassis_alarms : Failed
Total No of tests passed: 0
Total No of tests failed: 1
Overall Tests failed!!!

**Comment by ydnath** [comment]
https://github.com/Juniper/jsnapy/pull/347 has the fixes
